core/state/commontrie/trie.go

```diff
@@ -0,0 +1,71 @@
+package commontrie
+
+import (
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie"
+	"github.com/NethermindEth/juno/core/trie2"
+)
+
+type Trie interface {
+	Update(key, value *felt.Felt) error
+	Get(key *felt.Felt) (felt.Felt, error)
+	Hash() (felt.Felt, error)
+	HashFn() crypto.HashFn
+}
+
+// TrieAdapter wraps trie.Trie to implement commontrie.CommonTrie
+type TrieAdapter struct {
+	Trie *trie.Trie
+}
+
+func NewTrieAdapter(t *trie.Trie) *TrieAdapter {
+	return &TrieAdapter{Trie: t}
+}
+
+func (ta *TrieAdapter) Update(key, value *felt.Felt) error {
+	_, err := ta.Trie.Put(key, value)
+	return err
+}
+
+func (ta *TrieAdapter) Get(key *felt.Felt) (felt.Felt, error) {
+	value, err := ta.Trie.Get(key)
+	if err != nil {
+		return felt.Zero, err
+	}
+	return *value, nil
+}
+
+func (ta *TrieAdapter) Hash() (felt.Felt, error) {
+	root, err := ta.Trie.Root()
+	return *root, err
+}
+
+func (ta *TrieAdapter) HashFn() crypto.HashFn {
+	return ta.Trie.HashFn()
+}
+
+// Trie2Adapter wraps trie2.Trie to implement commontrie.CommonTrie
+type Trie2Adapter struct {
+	Trie *trie2.Trie
+}
```

Comment:
Similar comment to the one immediately above

Reply:
done